{
  "Severity": "HIGH",
  "UpdatedAt": "2021-11-08T21:43:18Z",
  "Package": {
    "Ecosystem": "PIP",
    "Name": "tensorflow-gpu"
  },
  "Advisory": {
    "DatabaseId": 5879,
    "Id": "GSA_kwCzR0hTQS0zcmN3LTlwOXgtNTgyds0W9w",
    "GhsaId": "GHSA-3rcw-9p9x-582v",
    "References": [
      {
        "Url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-3rcw-9p9x-582v"
      },
      {
        "Url": "https://nvd.nist.gov/vuln/detail/CVE-2021-41228"
      },
      {
        "Url": "https://github.com/tensorflow/tensorflow/commit/8b202f08d52e8206af2bdb2112a62fafbc546ec7"
      },
      {
        "Url": "https://github.com/advisories/GHSA-3rcw-9p9x-582v"
      }
    ],
    "Identifiers": [
      {
        "Type": "GHSA",
        "Value": "GHSA-3rcw-9p9x-582v"
      },
      {
        "Type": "CVE",
        "Value": "CVE-2021-41228"
      }
    ],
    "Description": "### Impact\nTensorFlow's `saved_model_cli` tool is vulnerable to a code injection as it [calls `eval` on user supplied strings](https://github.com/tensorflow/tensorflow/blob/87462bfac761435a46641ff2f10ad0b6e5414a4b/tensorflow/python/tools/saved_model_cli.py#L524-L550)\n  \n```python\ndef preprocess_input_exprs_arg_string(input_exprs_str):\n  ... \n  for input_raw in filter(bool, input_exprs_str.split(';')):\n    ...\n    input_key, expr = input_raw.split('=', 1)\n    input_dict[input_key] = eval(expr)\n  ...\n``` \n  \nThis can be used by attackers to run arbitrary code on the plaform where the CLI tool runs.\nHowever, given that the tool is always run manually, the impact of this is not severe. We have patched this by adding a `safe` flag which defaults to `True` and an explicit warning for users. \n\n### Patches\nWe have patched the issue in GitHub commit [8b202f08d52e8206af2bdb2112a62fafbc546ec7](https://github.com/tensorflow/tensorflow/commit/8b202f08d52e8206af2bdb2112a62fafbc546ec7).\n\nThe fix will be included in TensorFlow 2.7.0. We will also cherrypick this commit on TensorFlow 2.6.1, TensorFlow 2.5.2, and TensorFlow 2.4.4, as these are also affected and still in supported range.\n\n### For more information\nPlease consult [our security guide](https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md) for more information regarding the security model and how to contact us with issues and questions.\n\n### Attribution\nThis vulnerability has been reported by Omer Kaspi from Vdoo.",
    "Origin": "UNSPECIFIED",
    "PublishedAt": "2021-11-10T16:54:02Z",
    "Severity": "HIGH",
    "Summary": "Code injection in `saved_model_cli`",
    "UpdatedAt": "2023-02-02T05:04:02Z",
    "WithdrawnAt": "",
    "CVSS": {
      "Score": 7.5,
      "VectorString": "CVSS:3.1/AV:L/AC:H/PR:H/UI:N/S:C/C:H/I:H/A:H"
    }
  },
  "Versions": [
    {
      "FirstPatchedVersion": {
        "Identifier": "2.6.1"
      },
      "VulnerableVersionRange": "\u003e= 2.6.0, \u003c 2.6.1"
    },
    {
      "FirstPatchedVersion": {
        "Identifier": "2.4.4"
      },
      "VulnerableVersionRange": "\u003c 2.4.4"
    },
    {
      "FirstPatchedVersion": {
        "Identifier": "2.5.2"
      },
      "VulnerableVersionRange": "\u003e= 2.5.0, \u003c 2.5.2"
    }
  ]
}